repo_processing/dulwich/extract.py

```diff
@@ -0,0 +1,146 @@
+            commit_dict['blob_id'] = change.old.sha.decode()
+            commit_dict['path'] = path + "/" + change.old.path.decode()
+        else:
+            commit_dict['add'], commit_dict['delete'] = count_diff(repo[change.old.sha], repo[change.new.sha])
+            commit_dict['blob_id'] = change.new.sha.decode()
+            commit_dict['path'] = path + "/" + change.new.path.decode()
+    except UnicodeDecodeError as e:
+        pass
+    return commit_dict
+
+
+def process_walker(repo: Repo) -> List[Dict]:
+    """
+    Function processes WalkEntries mapping the changes info into list of dictionaries of:
+    List of: {author, commit_id, path, blob_id, change: add, delete}
+
+    :param repo: Repository instance.
+
+    :return: Returns list of mapped commits info.
+    """
+    walker = repo.get_walker()
+    multiple_commits = list()
+    for entry in walker:
+        commit_dict = {'author': entry.commit.author.decode(), 'commit_id': entry.commit.id.decode()}
+        for change in entry.changes():
+            if type(change) is not TreeChange:
+                continue
+            commit_dict = process_change_diff(change, commit_dict, repo, repo.path)
+            multiple_commits.append(commit_dict)
+    return multiple_commits
+
+
+def clone_or_instantiate(path: str) -> Repo:
+    """
+    Function cloned repo, if it is not existed, otherwise instantiate the existing one.
+    :param: The path in local directory or url to GitHub repository.
+
+    :return: Repository instance.
+    """
+    return Repo(path[path.rfind('/') + 1:]) if os.path.exists(path[path.rfind('/') + 1:]) else clone(path)
+
+
+def process_repo(path: str) -> List[Dict]:
+    """
+    Function starts to process repository by a given path.
+    :param path: The path in local directory or url to GitHub repository.
+
+    :return: Returns list of mapped commits info.
+    """
+    repo = clone_or_instantiate(path)
+    return process_walker(repo)
+
+
+def parse_repos_list(url_file: str = "https://gist.githubusercontent.com/EgorBu/823d01e7f12cb18495891fb752398037/raw"
+                                     "/8b793d61498b76759a552f8f259c61661dc427ff/repos.txt") -> List[Dict]:
+    """
+    Function downloads .txt file that contains info about processing repositories.
+    :param url_file: URL to the file.
+
+    :return: Returns list of dictionaries with values: url, invitation, stars, language.
+    """
+    r = requests.get(url_file)
+    lines = r.text.replace('\t', '').split('\n')[1:]
+    list_repos = list()
+    for i in range(0, len(lines), 4):
+        d = {
+            "url": "https://" + lines[i] if "https://" not in lines[i] else lines[i],
+            "invitation": lines[i + 1],
+            "stars": lines[i + 2],
+            "language": lines[i + 3]
+        }
+        list_repos.append(d)
+
+    return list_repos
+
+
+def save_file(multiple_commits: List[Dict], output_path: str = '') -> bool:
```

Comment:
let's remove logic related to `try...except` - and function shouldn't return anything